extensions/Worker.Extensions.ServiceBus/src/ServiceBusReceivedMessageConverter.cs

```diff
@@ -0,0 +1,44 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Azure.Messaging.ServiceBus;
+using Azure.Messaging.ServiceBus.Primitives;
+using Microsoft.Azure.Functions.Worker.Converters;
+using Microsoft.Azure.Functions.Worker.Core;
+
+namespace Microsoft.Azure.Functions.Worker;
+
+internal class ServiceBusReceivedMessageConverter : IInputConverter
+{
+    public ValueTask<ConversionResult> ConvertAsync(ConverterContext context)
+    {
+        ConversionResult result = context?.Source switch
+        {
+            ModelBindingData binding => ConversionResult.Success(ConvertToServiceBusReceivedMessage(binding)),
+            CollectionModelBindingData collection when context.TargetType.IsArray => ConversionResult.Success(collection.ModelBindingDataArray.Select(ConvertToServiceBusReceivedMessage).ToArray()),
```

Comment:
Need to do end-to-end tests for collections.

Reply:
This doesn't seem to work as expected - for one thing when ParameterBindingData comes into play it seems to wipe out the named string bindings. Is this something that needs to be handled here now? How can we detect the parameter name?

The other issue is that I am always getting back a ModelBindingData rather than a CollectionModelBindingData when binding to an array. I would have expected the ParameterBindingData converter defined in the extension to automatically handle collections, but that does not appear to happen. I tried defining an explicit converter for SBMessage array to ParameterBindingData array but that doesn't seem to work either.

@liliankasem could use some guidance here.

Reply:
So I think I answered my own question on the collection handling. I forgot that the isolated model requires setting the IsBatched trigger property.

I see that the ConverterContext has access to the parameter name and the binding contract so I can handle the named bindings myself if needed, but would still like to confirm that this is the recommendation here.

Reply:
Yes, IsBatched is used to assign the cardinality which is then used by the [Host](https://github.com/Azure/azure-functions-host/blob/8ceb05a89a4337f07264d4991545538a3e8b58a0/src/WebJobs.Script/Binding/GeneralScriptBindingProvider.cs#L70-L75) to create type `ParameterBindingData[]`. Example in [BlobInputAttribute](https://github.com/Azure/azure-functions-dotnet-worker/blob/0fd7bf6aef005e4b8a14874506bf7a8ad7ad73ef/extensions/Worker.Extensions.Storage.Blobs/src/BlobInputAttribute.cs#L33). Let me know if I am missing something specific to your scenario.